torchgen/api/types/signatures.py

```diff
@@ -349,23 +369,47 @@ def inner_call(self, *, reapply_views: Optional[bool] = None) -> str:
 
         arg_ctx = functionalization.outer_arguments(is_reverse=self.is_reverse)
         capture_ctx = functionalization.capture_arguments(
-            self.g.view.func, is_reverse=self.is_reverse
+            self.g.view.func,
+            is_reverse=self.is_reverse,
+            need_reapply_views=self.respects_reapply_views_tls,
         )
         full_ctx = arg_ctx + capture_ctx
 
         assert self.g.view_copy is not None
         call_bindings = functionalization.inner_arguments(
             self.g.view_copy.func, is_reverse=self.is_reverse
         )
+
+        # We might need to hard-code reapply_views=False. If so, remove it from the list
+        # of bindings and keep track of where it was for "false" insertion later.
+        # (See note above for _VIEWS_THAT_REQUIRE_SCATTER_INVERSE)
+        rv_index = None
+        if not self.respects_reapply_views_tls:
+            rv_index = call_bindings.index(functionalization.reapply_views_binding)
+            call_bindings.remove(functionalization.reapply_views_binding)
+
         maybe_index = functionalization.inner_call_index(self.g.view_copy.func)
         call_exprs = [
             e.expr for e in translate.translate(full_ctx, call_bindings, method=False)
         ]
+
+        if rv_index is not None:
+            call_exprs.insert(rv_index, "false")
+
         if not self.is_reverse and maybe_index is not None:
             return f'{inner_call_name}({", ".join(call_exprs)})[{maybe_index.name}];'
         else:
             return f'{inner_call_name}({", ".join(call_exprs)});'
 
+    @property
+    def respects_reapply_views_tls(self) -> bool:
+        # See note above for _VIEWS_THAT_REQUIRE_SCATTER_INVERSE
+        return not (
+            self.is_reverse
+            and self.g.view.func.name.unambiguous_name()
+            in _VIEWS_THAT_REQUIRE_SCATTER_INVERSE
```

Comment:
hmmm... just thinking - would it be simpler if instead we just update this [line](https://github.com/pytorch/pytorch/blob/main/torchgen/gen_functionalization_type.py#L340) to:
```
auto reapply_views = {'false;' if f.func.name.unambiguous_name() in _VIEWS_THAT_REQUIRE_SCATTER_INVERSE else 'at::functionalization::impl::getFunctionalizationReapplyViewsTLS();' }
```

Reply:
So I originally tried exactly that, but I found that I still need to query the TLS for the [forward part of codegen](https://github.com/pytorch/pytorch/blob/b062ea38039234c80404a8f5f4d5a93c4cb9832d/torchgen/gen_functionalization_type.py#L343-L347).

Reply:
ah good catch